src/utilities/bcl/BCLComponent.cpp

```diff
@@ -76,21 +76,32 @@ BCLComponent::BCLComponent(const openstudio::path& dir) : m_directory(dir) {
         std::string units = componentElement.child("units").text().as_string();
 
         if (datatype == "float") {
+          double castVal;
+          try {
+            castVal = boost::lexical_cast<double>(value);
+          } catch (const boost::bad_lexical_cast&) {
+            LOG_AND_THROW("Please double check your XML, you have an attribute named '" << name << "' with a datatype of '" << datatype
+                                                                                        << "' but the value isn't a double: '" << value << "'.");
+          }
           if (units.empty()) {
-            Attribute attr(name, boost::lexical_cast<double>(value));
-            m_attributes.push_back(attr);
+            m_attributes.emplace_back(name, castVal);
```

Comment:
While I'm at it, avoid unecessary copy/move and create object directly in vector via emplace_back